source/ed/osm2ed.cpp

```diff
@@ -970,71 +966,100 @@ void OSMCache::flag_nodes() {
+        params.poi_types.find("amenity:bicycle_rental") == params.poi_types.end()) {
+        throw std::invalid_argument("The 2 POI types id=amenity:parking and "
+                                    "id=amenity:bicycle_rental must be defined");
+    }
+
+    for (const pt::ptree::value_type& json_rule: poi_root.get_child("rules")) {
+        params.rules.emplace_back();
+        auto& rule = params.rules.back();
+        rule.poi_type_id = json_rule.second.get<std::string>("poi_type_id");
+        if (params.poi_types.find(rule.poi_type_id) == params.poi_types.end()) {
+            throw std::invalid_argument("Using an undefined POI type id (" + rule.poi_type_id + ") in rules");
+        }
+        for (const pt::ptree::value_type& json_filter: json_rule.second.get_child("osm_tags_filters")) {
+            rule.osm_tag_filters[json_filter.second.get<std::string>("key")] =
+                                                    json_filter.second.get<std::string>("value");
+        }
+    }
+    return params;
 }
 
 int main(int argc, char** argv) {
     navitia::init_app();
     auto logger = log4cplus::Logger::getInstance("log");
     pt::ptime start;
-    std::string input, connection_string;
-    std::vector<std::string> poi_types;
-    std::vector<std::string> default_poi_types = {
-            "amenity:college=école",
-            "amenity:university=université",
-            "amenity:theatre=théâtre",
-            "amenity:hospital=hôpital",
-            "amenity:post_office=bureau de poste",
-            "amenity:bicycle_rental=station vls",
-            "amenity:bicycle_parking=Parking vélo",
-            "amenity:parking=Parking",
-            "amenity:police=Police, Gendarmerie",
-            "amenity:townhall=Mairie",
-            "leisure:garden=Jardin",
-            "leisure:park=Zone Parc. Zone verte ouverte, pour déambuler. habituellement municipale"
-    };
+    std::string input, connection_string, json_poi_types;
+
     po::options_description desc("Allowed options");
     desc.add_options()
         ("version,v", "Show version")
         ("help,h", "Show this message")
-        ("input,i", po::value<std::string>(&input)->required(), "Input file (must be an OSM one)")
+        ("input,i", po::value<std::string>(&input),
+                    "Input file (must be an OSM one). "
+                    "If none provided, will check that json poi-type param (if provided) is correct.")
         ("connection-string", po::value<std::string>(&connection_string)->required(),
              "Database connection parameters: host=localhost user=navitia"
              " dbname=navitia password=navitia")
-        ("poi-type,p", po::value<std::vector<std::string>>(&poi_types)->default_value(default_poi_types, ""),
-                       "a poi_type like amenity:college with optionally a name separated with an equals, "
-                       "ex: amenity:college=école");
```

Comment:
That'll be problematic to stop supporting the old way. I think the old way should generate a filter to have the same behavior as before.